camel/societies/role_playing.py

```diff
@@ -159,25 +240,64 @@ def __init__(
                     (user_role_name, RoleType.USER),
                 ],
             ))
+        return init_assistant_sys_msg, init_user_sys_msg, sys_msg_meta_dicts
 
-        self.assistant_agent: ChatAgent = ChatAgent(
+    def init_agents(self, init_assistant_sys_msg: BaseMessage,
+                    assistant_agent_kwargs: Optional[Dict],
+                    init_user_sys_msg: BaseMessage,
+                    user_agent_kwargs: Optional[Dict],
+                    output_language: Optional[str]):
+        r"""Initialize assistant and user agents with their system messages.
+
+        Args:
+            init_assistant_sys_msg (BaseMessage): Assistant agent's initial
+                system message.
+            assistant_agent_kwargs (Dict, optional): Additional arguments to
+                pass to the assistant agent.
+            init_user_sys_msg (BaseMessage): User agent's initial system
+                message.
+            user_agent_kwargs (Dict, optional): Additional arguments to
+                pass to the user agent.
+            output_language (str, optional): The language to be output by the
+                agents.
+        """
+        self.assistant_agent = ChatAgent(
             init_assistant_sys_msg,
-            model_type,
+            self.model_type,
             output_language=output_language,
             **(assistant_agent_kwargs or {}),
         )
         self.assistant_sys_msg = self.assistant_agent.system_message
```

Comment:
@lightaime this is hacking into object's private fields. Later we need to either remove this, or add a property `system_message` to `ChatAgent ` and rename the field `system_message` into private `_system_message`.